tools/cmake/README.md

```diff
@@ -82,27 +82,57 @@ If you're running CMake directly, you may need to pass it in yourself as follows
 cmake -S <source-dir> -B <build-dir> -DCMAKE_BUILD_TYPE=debug
 ```
 
-## How to build a greentea test
-
-Install prerequisites suggested in the previous section and follow the below steps to build:
-* Set your current directory to the test suite directory
-
-* Run the following command for the configuration CMake module to be generated
-  ```
-  mbedtools configure -t <TOOLCHAIN> -m <MBED_TARGET> --mbed-os-path /path/to/mbed-os
-  ```
-* Build the test binary with the full profile
+## How to build and run greentea tests
+
+### Set up
+Install the [mbed-hosts-tests](https://pypi.org/project/mbed-host-tests/) python package.
+
+Also install the prerequisites suggested in the previous section.
+
+### Configuring the build system
+1. Set your current directory to the mbed-os root.
+1. Connect your target and detect it using `mbed-tools`
+   ```
+   mbedtools detect
+   ```
+   _Make a note of the mount point and serial port. We will need to pass them to CMake in a later step._
+
+1. Generate an mbed_config.cmake configuration file for your target using `mbed-tools`
+   ```
+   mbedtools configure -t <TOOLCHAIN> -m <MBED_TARGET> --mbed-os-path . --output-dir __build
+   ```
+1. Generate the build system with CMake, configuring it to build greentea tests.
+   Here you will pass the serial port and mount point you noted in an earlier
+   step
+   ```
+   cmake -S . -B __build -G Ninja -DBUILD_GREENTEA_TESTS=ON -DCMAKE_CTEST_ARGUMENTS="--output-on-failure;-V" -DMBED_HTRUN_ARGUMENTS="-p;<serial-port-from-detect>;-d;<mount-point-from-detect>"
+   ```
+   The `MBED_HTRUN_ARGUMENTS` argument is a semicolon separated list of arguments
+   to forward to `mbedhtrun`. You can pass any of the optional flags you would
+   usually pass to `mbedhtrun` by adding them to this list.
```

Comment:
Worth noting that each argument is already quoted by CTest so the user should not add quotation marks or escape any characters.
For example, a mount point has a space inside, in which case `-DMBED_HTRUN_ARGUMENTS="-p;/dev/tty.usbmodem14402;-d;/Volumes/NuMicro MCU"` just works, but `-DMBED_HTRUN_ARGUMENTS="-p;/dev/tty.usbmodem14402;-d;/Volumes/NuMicro\ MCU"` causes the backslash to be treated as a literal character, and `-DMBED_HTRUN_ARGUMENTS="-p;/dev/tty.usbmodem14402;-d;'/Volumes/NuMicro\ MCU'"` causes the single quotes to be treated as literal characters.